### PR TITLE
overlay: Export metadata for container and image in docker-inspect

### LINKED ISSUE #14099: overlay: Export image and container metadata through docker-inspect
Now devicemapper graph driver exports enough information about image/container through docker-inspect so that external tools can make use of that information and mount image/container for scanning or other operations.

Export similar metadata for overlay graphdriver also so that same tools can work with overlay backend too.

### PR DESCRIPTION
**Pull request by rhvgoyal**
Fixes #14099 

Export metadata for container and image in docker-inspect when overlay
graphdriver is in use. Right now it is done only for devicemapper graph
driver.

Signed-off-by: Vivek Goyal vgoyal@redhat.com

### CODE REVIEW

File: daemon/graphdriver/overlay/overlay.go [hunk 1]
```diff
@@ -168,7 +168,35 @@ func (d *Driver) Status() [][2]string {
...
+	lowerId, err := ioutil.ReadFile(path.Join(dir, "lower-id"))
+	if err != nil {
+		return nil, err
+	}
+
+	lowerDir := path.Join(d.dir(string(lowerId)), "root")
+	upperDir := path.Join(dir, "upper")
+	workDir := path.Join(dir, "work")
+	mergedDir := path.Join(dir, "merged")
+	metadata["lowerDir"] = lowerDir
```
Comment (LK4D4) [inline daemon/graphdriver/overlay/overlay.go]: any particular reason for using camelcase with lower first? in devicemapper simple CamelCase is used.

Comment (rhvgoyal) [inline daemon/graphdriver/overlay/overlay.go]: @LK4D4 

That's a good point. I did not pay attention to it. As other fields in docker inspect output seem to be CamleCase, I will change these as well.

**Comment by rhvgoyal** [comment]
@LK4D4 refreshed the patch with CamelCase. PTAL.

File: daemon/graphdriver/overlay/overlay.go [hunk 2]
```diff
@@ -168,7 +168,35 @@ func (d *Driver) Status() [][2]string {
...
+		metadata["RootDir"] = rootDir
+		return metadata, nil
+	}
+
+	lowerId, err := ioutil.ReadFile(path.Join(dir, "lower-id"))
+	if err != nil {
+		return nil, err
+	}
+
+	lowerDir := path.Join(d.dir(string(lowerId)), "root")
```
Comment (LK4D4) [inline daemon/graphdriver/overlay/overlay.go]: I think you can assign directly to map without intermediate variables.

**Comment by rhvgoyal** [comment]
@LK4D4 Got rid of local variables and directly assigned to map.

**Comment by LK4D4** [comment]
@rhatdan Thanks! LGTM

**Comment by rhvgoyal** [comment]
@LK4D4 I am @rhvgoyal and not @rhatdan. :-)

**Comment by LK4D4** [comment]
@rhvgoyal haha :) Those rh* guys...

**Comment by rhatdan** [comment]
I like to take credit for @rhvgoyal work though.  :^)

**Comment by jfrazelle** [comment]
Cool thanks LGTM

**Pull request merged**